Automattic/wp-calypso
# Sites Management Page: Improve search input performance with lazyloading

Related to p1675986927280589-slack-C0347E545HR

## Proposed Changes

Improves search input performance on Sites Management Page by lazyloading `<SiteItemThumbnail />` and `<SitesEllipsisMenu />`.

Before this change, the first keystroke takes ~150-190ms to complete:

<img width="1127" alt="image" src="https://user-images.githubusercontent.com/36432/218800181-f06baf9c-1505-45c2-9afe-8969db0d60b7.png">

With this change, the first keystroke takes ~100-130ms to complete:

<img width="914" alt="image" src="https://user-images.githubusercontent.com/36432/218800585-656eda4c-eeb4-4b35-9398-840b1cdc4e8e.png">

We save on performance by deferring some rendering until the given site is in view.

Another approach I considered was to reintroduce `delaySearch` (https://github.com/Automattic/wp-calypso/pull/65760/files#r924953509). However, doing so introduces some delay between keystroke and render, which is an inferior user experience. Plus, it somewhat ignores the underlying problem.

Lastly, another alternative would be to only show 36 sites instead of 96. The first keystroke looks to be in the 100ms range:

<img width="1183" alt="image" src="https://user-images.githubusercontent.com/36432/218803322-48a86615-939f-41e2-b348-7fa964f761da.png">

```diff
diff --git a/client/sites-dashboard/components/sites-dashboard.tsx b/client/sites-dashboard/components/sites-dashboard.tsx
index ea0b4e4f95..71153aafdf 100644
--- a/client/sites-dashboard/components/sites-dashboard.tsx
+++ b/client/sites-dashboard/components/sites-dashboard.tsx
@@ -138,7 +138,7 @@ const ScrollButton = styled( Button, { shouldForwardProp: ( prop ) => prop !== '
 const SitesDashboardSitesList = createSitesListComponent();
 
 export function SitesDashboard( {
-	queryParams: { page = 1, perPage = 96, search, status = 'all' },
+	queryParams: { page = 1, perPage = 36, search, status = 'all' },
 }: SitesDashboardProps ) {
 	const { __, _n } = useI18n();
 	const { data: allSites = [], isLoading } = useSiteExcerptsQuery();

```

## Testing Instructions

1. Navigate to `/sites`.
2. Test a search in List view and verify that it feels snappier.
3. Test a search in Grid view and verify that it feels snappier.
